diff --git a/dist/NodeFileSystem.js b/dist/NodeFileSystem.js
index a532e4f58f974fb90d60c1f5e710ff02f4856bbb..b4f2962dc17596e56df3055a9cf6fcf85acc20c1 100644
--- a/dist/NodeFileSystem.js
+++ b/dist/NodeFileSystem.js
@@ -207,7 +207,7 @@ const makeFile = /*#__PURE__*/(() => {
     write(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.map(nodeWrite(this.fd, buffer, undefined, undefined, this.append ? undefined : Number(position)), bytesWritten => {
+        return Effect.map(nodeWrite(this.fd, buffer, 0, buffer.length, this.append ? undefined : Number(position)), bytesWritten => {
           const sizeWritten = FileSystem.Size(bytesWritten);
           if (!this.append) {
             this.position = position + sizeWritten;
@@ -219,7 +219,7 @@ const makeFile = /*#__PURE__*/(() => {
     writeAllChunk(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.flatMap(nodeWriteAll(this.fd, buffer, undefined, undefined, this.append ? undefined : Number(position)), bytesWritten => {
+        return Effect.flatMap(nodeWriteAll(this.fd, buffer, 0, buffer.length, this.append ? undefined : Number(position)), bytesWritten => {
           if (bytesWritten === 0) {
             return Effect.fail(Error.systemError({
               module: "FileSystem",
diff --git a/src/NodeFileSystem.ts b/src/NodeFileSystem.ts
index 8be146df34954b63b996aac202f74aa4003a4dad..83843952310df040ecde2b89b82cfaa1d8258d30 100644
--- a/src/NodeFileSystem.ts
+++ b/src/NodeFileSystem.ts
@@ -341,7 +341,7 @@ const makeFile = (() => {
       return Effect.suspend(() => {
         const position = this.position
         return Effect.map(
-          nodeWrite(this.fd, buffer, undefined, undefined, this.append ? undefined : Number(position)),
+          nodeWrite(this.fd, buffer, 0, buffer.length, this.append ? undefined : Number(position)),
           (bytesWritten) => {
             const sizeWritten = FileSystem.Size(bytesWritten)
             if (!this.append) {
@@ -357,7 +357,7 @@ const makeFile = (() => {
       return Effect.suspend(() => {
         const position = this.position
         return Effect.flatMap(
-          nodeWriteAll(this.fd, buffer, undefined, undefined, this.append ? undefined : Number(position)),
+          nodeWriteAll(this.fd, buffer, 0, buffer.length, this.append ? undefined : Number(position)),
           (bytesWritten) => {
             if (bytesWritten === 0) {
               return Effect.fail(
