diff --git a/dist/NodeFileSystem.js b/dist/NodeFileSystem.js
index a94f50e..597a676 100644
--- a/dist/NodeFileSystem.js
+++ b/dist/NodeFileSystem.js
@@ -232,7 +232,7 @@ const makeFile = /*#__PURE__*/(() => {
     write(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"), nodePosition => Effect.map(nodeWrite(this.fd, buffer, undefined, undefined, nodePosition), bytesWritten => {
+        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"), nodePosition => Effect.map(nodeWrite(this.fd, buffer, 0, buffer.length, nodePosition), bytesWritten => {
           if (!this.append) {
             this.position = position + BigInt(bytesWritten);
           }
@@ -243,7 +243,7 @@ const makeFile = /*#__PURE__*/(() => {
     writeAllChunk(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"), nodePosition => Effect.flatMap(nodeWriteAll(this.fd, buffer, undefined, undefined, nodePosition), bytesWritten => {
+        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"), nodePosition => Effect.flatMap(nodeWriteAll(this.fd, buffer, 0, buffer.length, nodePosition), bytesWritten => {
           if (bytesWritten === 0) {
             return Effect.fail(Error.systemError({
               module: "FileSystem",
diff --git a/src/NodeFileSystem.ts b/src/NodeFileSystem.ts
index 19fabb0..4aed757 100644
--- a/src/NodeFileSystem.ts
+++ b/src/NodeFileSystem.ts
@@ -371,7 +371,7 @@ const makeFile = (() => {
           this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"),
           (nodePosition) =>
             Effect.map(
-              nodeWrite(this.fd, buffer, undefined, undefined, nodePosition),
+              nodeWrite(this.fd, buffer, 0, buffer.length, nodePosition),
               (bytesWritten) => {
                 if (!this.append) {
                   this.position = position + BigInt(bytesWritten)
@@ -390,7 +390,7 @@ const makeFile = (() => {
           this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"),
           (nodePosition) =>
             Effect.flatMap(
-              nodeWriteAll(this.fd, buffer, undefined, undefined, nodePosition),
+              nodeWriteAll(this.fd, buffer, 0, buffer.length, nodePosition),
               (bytesWritten) => {
                 if (bytesWritten === 0) {
                   return Effect.fail(
