--- a/dist/NodeFileSystem.js
+++ b/dist/NodeFileSystem.js
@@ -233,7 +233,7 @@
     write(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"), nodePosition => Effect.map(nodeWrite(this.fd, buffer, undefined, undefined, nodePosition), bytesWritten => {
+        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"), nodePosition => Effect.map(nodeWrite(this.fd, buffer, 0, buffer.length, nodePosition), bytesWritten => {
           if (!this.append) {
             this.position = position + BigInt(bytesWritten);
           }
@@ -244,7 +244,7 @@
     writeAllChunk(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"), nodePosition => Effect.flatMap(nodeWriteAll(this.fd, buffer, undefined, undefined, nodePosition), bytesWritten => {
+        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"), nodePosition => Effect.flatMap(nodeWriteAll(this.fd, buffer, 0, buffer.length, nodePosition), bytesWritten => {
           if (bytesWritten === 0) {
             return Effect.fail(Error.systemError({
               module: "FileSystem",
--- a/src/NodeFileSystem.ts
+++ b/src/NodeFileSystem.ts
@@ -372,7 +372,7 @@
           this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"),
           (nodePosition) =>
             Effect.map(
-              nodeWrite(this.fd, buffer, undefined, undefined, nodePosition),
+              nodeWrite(this.fd, buffer, 0, buffer.length, nodePosition),
               (bytesWritten) => {
                 if (!this.append) {
                   this.position = position + BigInt(bytesWritten)
@@ -391,7 +391,7 @@
           this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"),
           (nodePosition) =>
             Effect.flatMap(
-              nodeWriteAll(this.fd, buffer, undefined, undefined, nodePosition),
+              nodeWriteAll(this.fd, buffer, 0, buffer.length, nodePosition),
               (bytesWritten) => {
                 if (bytesWritten === 0) {
                   return Effect.fail(
