diff --git a/dist/NodeFileSystem.js b/dist/NodeFileSystem.js
index 532c1c7987afef3e27249000cb6c9782df4d1952..c9767b8b551252f4ce3a45973a91e20279b6d19c 100644
--- a/dist/NodeFileSystem.js
+++ b/dist/NodeFileSystem.js
@@ -231,7 +231,7 @@ const makeFile = /*#__PURE__*/(() => {
     write(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"), nodePosition => Effect.map(nodeWrite(this.fd, buffer, undefined, undefined, nodePosition), bytesWritten => {
+        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"), nodePosition => Effect.map(nodeWrite(this.fd, buffer, 0, buffer.length, nodePosition), bytesWritten => {
           if (!this.append) {
             this.position = position + BigInt(bytesWritten);
           }
@@ -242,7 +242,7 @@ const makeFile = /*#__PURE__*/(() => {
     writeAllChunk(buffer) {
       return Effect.suspend(() => {
         const position = this.position;
-        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"), nodePosition => Effect.flatMap(nodeWriteAll(this.fd, buffer, undefined, undefined, nodePosition), bytesWritten => {
+        return Effect.flatMap(this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"), nodePosition => Effect.flatMap(nodeWriteAll(this.fd, buffer, 0, buffer.length, nodePosition), bytesWritten => {
           if (bytesWritten === 0) {
             return Effect.fail(Error.systemError({
               module: "FileSystem",
diff --git a/src/NodeFileSystem.ts b/src/NodeFileSystem.ts
index 586142507ce0d0cceb793afbcedf066f361c4943..cc679a71013bd137b4ffa83f1f5ebd0606ef0396 100644
--- a/src/NodeFileSystem.ts
+++ b/src/NodeFileSystem.ts
@@ -370,7 +370,7 @@ const makeFile = (() => {
           this.append ? Effect.succeed(undefined) : positionToNumber(position, "write"),
           (nodePosition) =>
             Effect.map(
-              nodeWrite(this.fd, buffer, undefined, undefined, nodePosition),
+              nodeWrite(this.fd, buffer, 0, buffer.length, nodePosition),
               (bytesWritten) => {
                 if (!this.append) {
                   this.position = position + BigInt(bytesWritten)
@@ -389,7 +389,7 @@ const makeFile = (() => {
           this.append ? Effect.succeed(undefined) : positionToNumber(position, "writeAll"),
           (nodePosition) =>
             Effect.flatMap(
-              nodeWriteAll(this.fd, buffer, undefined, undefined, nodePosition),
+              nodeWriteAll(this.fd, buffer, 0, buffer.length, nodePosition),
               (bytesWritten) => {
                 if (bytesWritten === 0) {
                   return Effect.fail(
